File: trunk/examples/CBuilder/ExBrowsU.cpp
// ***** BEGIN LICENSE BLOCK *****
// * Version: MPL 1.1
// *
// * The contents of this file are subject to the Mozilla Public License Version
// * 1.1 (the "License"); you may not use this file except in compliance with
// * the License. You may obtain a copy of the License at
// * http://www.mozilla.org/MPL/
// *
// * Software distributed under the License is distributed on an "AS IS" basis,
// * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
// * for the specific language governing rights and limitations under the
// * License.
// *
// * The Original Code is TurboPower ShellShock
// *
// * The Initial Developer of the Original Code is
// * TurboPower Software
// *
// * Portions created by the Initial Developer are Copyright (C) 1996-2002
// * the Initial Developer. All Rights Reserved.
// *
// * Contributor(s):
// *
// * ***** END LICENSE BLOCK *****
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExBrowsU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "StAbout"
#pragma link "SsBase"
#pragma resource "*.dfm"
TMainForm *MainForm;
//---------------------------------------------------------------------------
__fastcall TMainForm::TMainForm(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::AboutBtnClick(TObject *Sender)
{
  StShellAbout1->Execute();
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::BrowseBtnClick(TObject *Sender)
{
  TStBrowseOptionsSet Options;
  Options.Clear();
  if (ComputerCB->Checked)
    Options << boBrowseForComputer;
  if (PrinterCB->Checked)
    Options << boBrowseForPrinter;
  if (DomainCB->Checked)
    Options << boDontGoBelowDomain;
  if (AncestorCB->Checked)
    Options << boReturnOnlyAncestors;
  if (DirectoriesCB->Checked)
    Options << boReturnOnlyDirs;
  if (ShowFilesCB->Checked)
    Options << boShowFiles;
  if (ShowEditCB->Checked)
    Options << boEditBox;
  StBrowser1->Options = Options;
  StBrowser1->RootFolder = RootEdit->Text;
  StBrowser1->SelectedFolder = SelFolderEdit->Text;
  if (StartFolderCombo->ItemIndex == -1 || StartFolderCombo->Text == "(None)")
    StBrowser1->SpecialRootFolder = sfNone;
  else
    StBrowser1->SpecialRootFolder = (TStSpecialRootFolder)
    (StartFolderCombo->Items->IndexOf(StartFolderCombo->Text) - 1);
  if (StBrowser1->Execute()) {
    DisplayNameLabel->Caption = StBrowser1->DisplayName;
    PathLabel->Caption = StBrowser1->Path;
    ImageIndexLabel->Caption = IntToStr(StBrowser1->ImageIndex);
  }
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::StBrowser1SelChanged(TObject *Sender)
{
  if (UpperCase(StBrowser1->SelectedFolder) == UpperCase("d:\\"))
    StBrowser1->OKEnabled = False;
  SelectedFolderLabel->Caption = StBrowser1->SelectedFolder;
}
//---------------------------------------------------------------------------
void __fastcall TMainForm::FormCreate(TObject *Sender)
{
  StartFolderCombo->Text = StartFolderCombo->Items->Strings[0];
}
//---------------------------------------------------------------------------

File: trunk/examples/Delphi/ExBrowsU.pas
(* ***** BEGIN LICENSE BLOCK *****
 * Version: MPL 1.1
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 1.1 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * The Original Code is TurboPower ShellShock
 *
 * The Initial Developer of the Original Code is
 * TurboPower Software
 *
 * Portions created by the Initial Developer are Copyright (C) 1996-2002
 * the Initial Developer. All Rights Reserved.
 *
 * Contributor(s):
 *
 * ***** END LICENSE BLOCK ***** *)

unit ExBrowsU;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, StAbout, StBrowsr, ComCtrls, StFormat, SsBase;

type
  TMainForm = class(TForm)
    AboutBtn: TButton;
    StShellAbout1: TStShellAbout;
    GroupBox1: TGroupBox;
    Label1: TLabel;
    BrowseBtn: TButton;
    StBrowser1: TStBrowser;
    DisplayNameLabel: TLabel;
    Label2: TLabel;
    PathLabel: TLabel;
    Label3: TLabel;
    ImageIndexLabel: TLabel;
    GroupBox2: TGroupBox;
    ComputerCB: TCheckBox;
    PrinterCB: TCheckBox;
    DomainCB: TCheckBox;
    AncestorCB: TCheckBox;
    DirectoriesCB: TCheckBox;
    Label4: TLabel;
    SelectedFolderLabel: TLabel;
    RootEdit: TEdit;
    Label5: TLabel;
    Label6: TLabel;
    SelFolderEdit: TEdit;
    Label7: TLabel;
    StartFolderCombo: TComboBox;
    ShowFilesCB: TCheckBox;
    ShowEditCB: TCheckBox;
    procedure AboutBtnClick(Sender: TObject);
    procedure BrowseBtnClick(Sender: TObject);
    procedure StBrowser1SelChanged(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  MainForm: TMainForm;
  Continue : Boolean;

implementation

{$R *.DFM}

procedure TMainForm.AboutBtnClick(Sender: TObject);
begin
  StShellAbout1.Execute;
end;

procedure TMainForm.BrowseBtnClick(Sender: TObject);
var
  Opt : TStBrowseOptionsSet;
begin
  Opt := [];
  if ComputerCB.Checked then
    Opt := Opt + [boBrowseForComputer];
  if PrinterCB.Checked then
    Opt := Opt + [boBrowseForPrinter];
  if DomainCB.Checked then
    Opt := Opt + [boDontGoBelowDomain];
  if AncestorCB.Checked then
    Opt := Opt + [boReturnOnlyAncestors];
  if DirectoriesCB.Checked then
    Opt := Opt + [boReturnOnlyDirs];
  if ShowFilesCB.Checked then
    Opt := Opt + [boShowFiles];
  if ShowEditCB.Checked then
    Opt := Opt + [boEditBox];
  with StBrowser1 do begin
    Options := Opt;
    RootFolder := RootEdit.Text;
    SelectedFolder := SelFolderEdit.Text;
    if (StartFolderCombo.ItemIndex = -1) or (StartFolderCombo.Text = '(None)') then
      SpecialRootFolder := sfNone
    else
      SpecialRootFolder := TStSpecialRootFolder(
        StartFolderCombo.Items.IndexOf(StartFolderCombo.Text) + 1);
    if Execute then begin
      DisplayNameLabel.Caption := DisplayName;
      PathLabel.Caption := Path;
      ImageIndexLabel.Caption := IntToStr(ImageIndex);
    end;
  end;
end;

procedure TMainForm.StBrowser1SelChanged(Sender: TObject);
begin
  with StBrowser1 do begin
    if UpperCase(SelectedFolder) = UpperCase('d:\') then
      OKEnabled := False;
    SelectedFolderLabel.Caption := SelectedFolder;
  end;
end;

procedure TMainForm.FormCreate(Sender: TObject);
begin
  StartFolderCombo.Text := StartFolderCombo.Items[0];
end;

end.
